Update table

Kaiserabbas opened the pull request:
# Table Updates, Addition, Deletion, Savpoint, RollBack, Commit.
- Added a column species of type string to your animals table. 
- Inserted the new data.
- Updated the animals table by setting the species column to unspecified.
- Updated the animals table by setting the species column to digimon.
- Updated the animals table by setting the species column to pokemon.
- Deleted all records in the animals table, then roll back the transaction.
- Created a savepoint for the transaction.
- Wrote queries to answer the specified questions:
  How many animals are there?
How many animals have never tried to escape?
What is the average weight of animals?
Who escapes the most, neutered or not neutered animals?
What is the minimum and maximum weight of each type of animal?

**After addition of species column**
![5](https://github.com/Kaiserabbas/vet_clinic/assets/123939543/302af394-2018-47f3-9a39-d6ec2acf509c)

**Update the animals table by setting the species column to unspecified.** 
![6](https://github.com/Kaiserabbas/vet_clinic/assets/123939543/59cf739b-3380-4c93-b11a-864d96750d3e)

**Updated the animals table by setting the species column to digimon.
Updated the animals table by setting the species column to pokemon.**
![7](https://github.com/Kaiserabbas/vet_clinic/assets/123939543/bdd2e738-86bb-453c-8145-9038cda37658)

**Deleted all records in the animals table, then roll back the transaction.**
![8](https://github.com/Kaiserabbas/vet_clinic/assets/123939543/31a9a1be-3d7f-4f07-a7ff-1ed7c2caa391)

**Created a savepoint for the transaction.**
![9](https://github.com/Kaiserabbas/vet_clinic/assets/123939543/f6869331-5c5c-45c0-9f03-ad8fc4b7433d)

**Wrote queries to answer the specified questions:**
![Screenshot 2023-09-26 092403](https://github.com/Kaiserabbas/vet_clinic/assets/123939543/6c11e714-4505-4010-9319-06363e9bdb8e)

Kaiserabbas committed:
Update schema.sql

Kaiserabbas committed:
Update data.sql

Kaiserabbas committed:
Update queries.sql

Kaiserabbas committed:
Merge branch 'dev' into update-table

shahadat3669 commented on data.sql:
```diff
@@ -14,16 +14,16 @@ VALUES ('Pikachu', '2021-01-07', 15.04, false, 1),
+-- Adding new animals after the addition os species column
 INSERT INTO animals (name, date_of_birth, weight_kg, neutered, escape_attempts, species) VALUES
 ('Charmander', '2020-02-08', -11, false, 0, 'Lizard'),
 ('Plantmon', '2021-11-15', -5.7, true, 2, 'Dog'),
 ('Squirtle', '1993-04-02', -12.13, false, 3, 'Turtle'),
 ('Angemon', '2005-06-12', -45, true, 1, 'Cat'),
 ('Boarmon', '2005-06-07', 20.4, true, 7, 'Boar'),
-('Blossom', '1998-10-13', 17, true, 3, 'Cat'),
-('Ditto', '2022-05-14', 22, true, 4, 'Rabbit');
+('Blossom', '1998-10-13', 17, true, 3, 'Rabbit'),
+('Ditto', '2022-05-14', 22, true, 4, 'Cat');
```
- [ ] 👍 Great job adding the data! However, it's important to ensure that the data aligns with the project requirements. I noticed that you added some values in the `species` column that were not specified in the project requirements. To maintain consistency with the project requirements, kindly review the data you've added and remove any species values that were randomly entered.
- [ ] Additionally, if you make any updates to the data, please include new queries and screenshots after the updates, if necessary. This will help ensure that your data is in accordance with the project's specifications. Keep up the good work! 👍📊📸

shahadat3669 commented on queries.sql:
```diff
@@ -1,24 +1,94 @@
+-- Delete all records in the 'animals' table
+DELETE FROM animals;
+-- Roll back the transaction to undo the deletion
+ROLLBACK;
+--  verify if all records in the animals table still exists
+SELECT * FROM animals;
+
+
+-- Create a savepoint for the transaction after deletion.
+-- Start a transaction
+BEGIN;
+-- Delete all animals born after Jan 1st, 2022
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+-- Create a savepoint
+SAVEPOINT my_savepoint;
+-- Update all animals' weight to be their weight multiplied by -1
+UPDATE animals SET weight_kg = weight_kg * -1;
+-- Rollback to the savepoint
+ROLLBACK TO SAVEPOINT my_savepoint;
+-- Update all animals' weights that are negative to be their weight multiplied by -1
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+-- Commit the transaction
+COMMIT;
+
+--Write the queries
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, MAX(escape_attempts) AS max_escape_attempts
+FROM animals
+GROUP BY neutered;
+
+SELECT species, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight
+FROM animals
+GROUP BY species;
+
+SELECT species, AVG(escape_attempts) AS avg_escape_attempts
+FROM animals
+WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
+GROUP BY species;
```
- [ ] As I mentioned previously, after updating data, please include new queries and screenshots, because currently, all screenshot has the wrong output. This will help ensure that your data is in accordance with the project's specifications. Keep up the good work! 👍📊📸

shahadat3669 reviewed (changes requested):
Hi @Kaiserabbas, 

Great job so far! 👏 You've done well, and you're almost there for the next project ⏭️, but there are a few issues 🐛 that you still need to work on.

<img  src="https://media2.giphy.com/media/m3864rBwwBTKMAbpn3/giphy.gif" />


## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ put correct SQL commands in the correct files 👍
✅ add a new column called species
✅ create queries to answer the  questions in transactions


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        <img src="https://img.shields.io/badge/github-%23121011.svg?style=for-the-badge&logo=github&logoColor=white" alt="GitHub">
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        <img src="https://img.shields.io/badge/linkedin-%230077B5.svg?style=for-the-badge&logo=linkedin&logoColor=white" alt="LinkedIn">
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        <img src="https://img.shields.io/badge/Slack-4A154B?style=for-the-badge&logo=slack&logoColor=white" alt="Slack">
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        <img src="https://img.shields.io/badge/Gmail-D14836?style=for-the-badge&logo=gmail&logoColor=white" alt="Email">
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Kaiserabbas committed:
Update data.sql

Kaiserabbas commented:
I have removed the species names add by name and changed it to null at the beginning and then all actions were done as per project requirements. I have re-taken the screenshots and updated the PR request accordingly.

DuaneDave reviewed (approved):
Hi @Kaiserabbas 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ All screenshots well updated
✔️ Data table is reflecting the proposed schema

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
